frontend/src/components/createUserModal/index.tsx

```diff
@@ -0,0 +1,194 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import TextField from '@material-ui/core/TextField';
+import FormGroup from '@material-ui/core/FormGroup';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Checkbox from '@material-ui/core/Checkbox';
+import FormLabel from '@material-ui/core/FormLabel';
+import Select from '@material-ui/core/Select';
+import MenuItem from '@material-ui/core/MenuItem';
+import Button from '@material-ui/core/Button';
+import Clear from '@material-ui/icons/Clear';
+
+import { createStaff } from '../../features/staffSlice';
+import { toDateStr } from '../../constants/global';
+
+import 'toastr/build/toastr.min.css';
+import createUserModalStyle from './createUserModal.module.scss';
+
+interface Props {
+  closeModalFunction: () => void;
+}
+
+interface TextField {
+  id: string;
+  name: string;
+  type: string;
+}
+
+const textFields: TextField[] = [
+  { id: 'firstName', name: 'First Name', type: 'text' },
+  { id: 'surname', name: 'Surname', type: 'text' },
+  { id: 'jobTitle', name: 'Job title', type: 'text' },
+  { id: 'contractedHours', name: 'Contracted hours', type: 'number' },
+  { id: 'pay', name: 'Pay', type: 'number' },
+];
+
+const CreateUserModal = ({ closeModalFunction }: Props) => {
+  const dispatch = useDispatch();
+  const [preferredDays, setPreferredDays] = useState<{
+    [key: string]: boolean;
+  }>({
+    Mon: false,
+    Tue: false,
+    Wed: false,
+    Thu: false,
+    Fri: false,
+    Sat: false,
+    Sun: false,
+  });
+
+  const [userInfo, setUserInfo] = useState<{ [key: string]: string | number }>({
```

Comment:
Can we be more specific about the type and either reuse StaffData or create a similar object to hold the data?
At the moment we can set the state to an (almost) arbitrary object with string or number values.

Reply:
Im gonna create a new type because staffdata doesnt quite fit